### ISSUE TITLE
Cursor offset when typing

### ISSUE THREAD

**Issue by JezuzStardust** [macvim-dev/macvim#970]
**Describe the bug**
The further I type on a line the more the cursor is offset away from the text (to the right). 

**To Reproduce**
Detailed steps to reproduce the behavior:
1. Run `mvim --clean`
2. Type `i` and start typing. 

**Expected behavior**

<img width="643" alt="Screenshot 2019-10-22 at 22 19 06" src="https://user-images.githubusercontent.com/31573403/67329726-8c671a00-f51b-11e9-91f5-ed181f058152.png">

The last letter being typed in the screenshot is the period but the cursor is way off. 

**Screenshots**
If applicable, copy/paste the text or add screenshots to help explain your problem.

**Environment (please complete the following information):**
 - Vim version: `VIM - Vi IMproved 8.1 (2018 May 18, compiled Oct 21 2019 08:38:12)
macOS version
Included patches: 1-2168
Compiled by Homebrew`
 - OS: macOS 10.15
 - Terminal: GUI

**Comment by eirnym** [comment]
@JezuzStardust thank you for an issue,

I can't reproduce it with version from GitHub. Could you check it as well in your environment?

**Comment by JezuzStardust** [comment]
@eirnym I just now tried to install from head (`brew uninstal macvim && brew install macvim --HEAD`. Is this the correct way to install the GitHub version or do I have to clone it and compile it myself?

With the commands I wrote above I get the same problem when using `mvim --clean` (and of course also if I actually use my .vimcr).

**Comment by eirnym** [comment]
yes, it's correct. GitHub binaries are official, I personally use them for many years.

**Comment by JezuzStardust** [comment]
OK, just tried it again using the binary this time. Still the same problem. I also noticed that the cursor resets to the correct position if the line wraps or if I start a new line of text.

**Comment by eirnym** [comment]
it's strange. I have no such problem at all (and haven't). I have tried to build MacVim from source and still no such problem. Could you investigate the problem further?

**Comment by JezuzStardust** [comment]
I managed to find the problem now! For some reason I had disabled the option "Use Core Text Renderer" under the preference menu. When I re-enabled that and restarted MacVim, the problem went away. I am not sure, but perhaps this is still something to investigate?

**Comment by ychin** [comment]
Huh that's odd. I would put this to investigate later but I would like to deprecate non-Core Text Renderer path (it doesn't work perfectly and has glitches) so likely not immediately. If there are other reasons why you can't use the Core Text renderer let me know.

**Comment by nblievers** [comment]
@JezuzStardust  thank you! This was driving me nuts.

**Comment by eirnym** [comment]
@ychin Good work! 

Snapshot 160 was fixed the issue, but font rendering is… slightly different (more spaces in between letters). I won't create another issue now  as you want to sunset the non-Core Text Rendering. Please, notify me if you need this issue
